java/flight/src/main/java/org/apache/arrow/flight/ArrowMessage.java

```diff
@@ -182,27 +169,20 @@ public HeaderType getMessageType() {
     return HeaderType.getHeader(message.headerType());
   }
 
-  public Message asSchemaMessage() {
```

Comment:
So this needs to be removed since it returns a FlatBuffer message, but would we need an alternative that returns the wrapper so someone could still get the FlatBuffer message?

Reply:
Potentially, I think given this refactoring that fact the no-one else needs the reference means we can expose it at a later time.

Reply:
Since it's a public API, I'm a little worried that someone might depend on this downstream. If we had
`public MessageMetadataResult asSchemaMessage()` then someone could still get the `Message` by calling `MessageMetadataResult.getMessage()` if needed right?

Reply:
added back.  I think in general we might be exposing too many public APIs without cause, but it is better to not make clients wait 2 release cycles if someone does in fact depend on this.

Reply:
>I think in general we might be exposing too many public APIs without cause

I agree. I'm not sure how this is used, but since it's already exposed publicly, we should be careful about changes.